benchmarks/base_classes.py

```diff
@@ -344,3 +347,72 @@ class T2IAdapterSDXLBenchmark(T2IAdapterBenchmark):
 
     def __init__(self, args):
         super().__init__(args)
+
+
+class BaseBenchmarkTestCase:
+    model_class = None
+    pretrained_model_name_or_path = None
+    model_class_name = None
+
+    def __init__(self):
+        super().__init__()
+
+    def get_result_filepath(self, suffix):
+        name = (
+            self.model_class_name
+            + "_"
+            + self.pretrained_model_name_or_path.replace("/", "_")
+            + "_"
+            + f"{suffix}.csv"
+        )
+        filepath = os.path.join(BASE_PATH, name)
+        return filepath
+
+
+class AutoencoderKLBenchmark(BaseBenchmarkTestCase):
+    model_class = AutoencoderKL
+
+    def __init__(self, pretrained_model_name_or_path, dtype, **kwargs):
+        super().__init__()
+        self.dtype = getattr(torch, dtype)
+        model = self.model_class.from_pretrained(pretrained_model_name_or_path, torch_dtype=self.dtype, **kwargs).eval()
+        model = model.to("cuda")
+        self.model = model
+        self.model_class_name = str(self.model.__class__.__name__)
+        self.pretrained_model_name_or_path = pretrained_model_name_or_path
+
+    @torch.no_grad
+    def run_decode(self, model, tensor):
+        _ = model.decode(tensor)
+
+    @torch.no_grad
+    def _test_decode(self, **kwargs):
+        batch = kwargs.get("batch")
+        height = kwargs.get("height")
+        width = kwargs.get("width")
+
+        tensor = torch.randn((batch, self.model.config.latent_channels, height, width), dtype=self.dtype, device="cuda")
+
+        time = benchmark_fn(self.run_decode, self.model, tensor)
+        memory = bytes_to_giga_bytes(torch.cuda.max_memory_reserved())
+        benchmark_info = BenchmarkInfo(time=time, memory=memory)
+        csv_dict = generate_csv_dict_model(
+            model_cls=self.model_class_name, ckpt=self.pretrained_model_name_or_path, benchmark_info=benchmark_info, **kwargs,
+        )
+        print(f"{self.model_class_name} decode - shape: {list(tensor.shape)}, time: {time}, memory: {memory}")
+        return csv_dict
+
+    def test_decode(self):
```

Comment:
Not needed for the first iteration but I would consider also including `model.compile()`.